mysql-test/suite/innodb/r/innodb-index.result

```diff
@@ -1947,3 +1947,63 @@ Warnings:
 Warning	1082	InnoDB: Table test/t contains 0 indexes inside InnoDB, which is different from the number of indexes 1 defined in the MariaDB 
 Warning	1082	InnoDB: Table test/t contains 0 indexes inside InnoDB, which is different from the number of indexes 1 defined in the MariaDB 
 DROP TABLE t;
+#
+#  MDEV-35115 Inconsistent Replace behaviour when multiple
+#               unique index exist
+#
+CREATE TABLE t1 (c1 NUMERIC UNSIGNED NOT NULL,
+c2 INT3 UNIQUE,
+c3 BIT(2) PRIMARY KEY)ENGINE=InnoDB;
+CREATE UNIQUE INDEX i1 ON t1(c1);
+INSERT INTO t1 (c1,c2,c3) VALUES (0,0,b'01');
+INSERT INTO t1 (c1,c2,c3) VALUES (1,1,b'10');
+FLUSH STATUS;
+SELECT * FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'HANDLER_%' AND VARIABLE_VALUE > 0;
+VARIABLE_NAME	VARIABLE_VALUE
+HANDLER_TMP_WRITE	24
+REPLACE INTO t1 (c1,c2,c3) VALUES (0,1,b'11');
+SELECT * FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'HANDLER_%' AND VARIABLE_VALUE > 0;
+VARIABLE_NAME	VARIABLE_VALUE
+HANDLER_COMMIT	1
+HANDLER_DELETE	2
```

Comment:
I had expected 0 or 1 `HANDLER_DELETE` for the `REPLACE` statement, instead of 2 or 1 like this test shows. What is the other `HANDLER_DELETE` for? Can you add an `--echo #` comment about it?